### PR TITLE
add template and dependent docs and resources on

### PR DESCRIPTION
**Pull request by seekdavidlee**
app service environment with azure sql, and vpn gateway @bmoore-msft

### CODE REVIEW

**Comment by seekdavidlee** [comment]
Hi @bmoore-msft - The error mentioned "There are no available subscriptions which we would place this environment in." - which I don't understand. Can you help?

**Comment by bmoore-msft** [comment]
Is this supported in US Gov cloud?

**Comment by seekdavidlee** [comment]
This is not supported on US Gov Cloud (but I don't see why not, the only limitation is because I don't have a US Gov Cloud specific subscription to test this on).

**Comment by bmoore-msft** [comment]
Where do we say it's not supported there? (in docs, etc)

**Comment by seekdavidlee** [comment]
I can update the docs to say it is not supported, and how do I tell this Azure DevOps to skip USGov test?

**Comment by bmoore-msft** [comment]
https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#cloud-support

**Comment by seekdavidlee** [comment]
Hi @bmoore-msft - Thanks for the link, it looks like we are good on a clean PR now. Thanks again!

File: 101-asev2-appservice-sql-vpngw/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,716 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
```
Comment (bmoore-msft) [inline 101-asev2-appservice-sql-vpngw/azuredeploy.json:4]: add metadata descriptions to all parameters

File: 101-asev2-appservice-sql-vpngw/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,716 @@
...
+            "defaultValue": "Standard_B2ms"
+        }
+    },
+    "variables": {
+        "location": "[parameters('location')]",
+        "vnetName": "[concat(parameters('stackName'), '-vnet')]",
+        "frontendSubnet": "frontend",
+        "backendSubnet": "backend",
+        "gatewaySubnet": "GatewaySubnet",
+        "sqlAdministratorLogin": "appadmin",
```
Comment (bmoore-msft) [inline 101-asev2-appservice-sql-vpngw/azuredeploy.json:50]: this must be a parameter

File: 101-asev2-appservice-sql-vpngw/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,716 @@
...
+        "numberOfWorkersFromWorkerPool": 1,
+        "nicName": "[concat( parameters('stackName'),'vmnic')]",
+        "privateDbEndpointName": "[concat( parameters('stackName'),'pvenp')]",
+        "privateDbDnsZoneName": "[parameters('privateDbDnsZoneName')]",
+        "privateDbDnsGroupName": "[concat(variables('privateDbEndpointName'),'/mydnsgroupname')]",
+        "privateHost1DnsZoneName": "[concat(parameters('stackName'),'.appserviceenvironment.net') ]",
+        "gatewayPublicIPName": "[concat( parameters('stackName'),'gwpip')]",
+        "gatewayName": "[concat( parameters('stackName'),'gw')]",
+        "vpnClientAddressPoolPrefix": "172.15.0.0/24",
+        "vmName": "[if(greater(length(parameters('stackName')),12), substring(parameters('stackName'),0,12) , parameters('stackName'))]"
```
Comment (bmoore-msft) [inline 101-asev2-appservice-sql-vpngw/azuredeploy.json:65]: remove the if() and just use substring()

File: 101-asev2-appservice-sql-vpngw/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,716 @@
...
+                        "name": "[variables('databaseEdition')]",
+                        "tier": "[variables('databaseEdition')]",
+                        "capacity": 5
+                    },
+                    "properties": {
+                        "edition": "[variables('databaseEdition')]",
+                        "collation": "[variables('databaseCollation')]",
+                        "requestedServiceObjectiveName": "[variables('databaseServiceObjectiveName')]"
+                    },
+                    "dependsOn": [
```
Comment (bmoore-msft) [inline 101-asev2-appservice-sql-vpngw/azuredeploy.json:212]: move all dependsOn above properties

File: 101-asev2-appservice-sql-vpngw/azuredeploy.parameters.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            "value": "GEN-GUID"
+        },
+        "aadUsername": {
+            "value": "upn"
+        },
+        "privateDbDnsZoneName": {
+            "value": "privatelink.database.windows.net"
+        },
+        "LoginUser": {
+            "value": "opsdev"
```
Comment (bmoore-msft) [inline 101-asev2-appservice-sql-vpngw/azuredeploy.parameters.json:18]: GEN-UNIQUE

File: 101-asev2-appservice-sql-vpngw/azuredeploy.parameters.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "stackName": {
+            "value": "GEN-UNIQUE"
+        },
+        "aadUserObjectId": {
+            "value": "GEN-GUID"
+        },
+        "aadUsername": {
+            "value": "upn"
```
Comment (bmoore-msft) [inline 101-asev2-appservice-sql-vpngw/azuredeploy.parameters.json:12]: GEN-UNIQUE

File: 101-asev2-appservice-sql-vpngw/azuredeploy.parameters.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        },
+        "privateDbDnsZoneName": {
+            "value": "privatelink.database.windows.net"
```
Comment (bmoore-msft) [inline 101-asev2-appservice-sql-vpngw/azuredeploy.parameters.json:15]: move this to a defaultValue on the parameter

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 101-asev2-appservice-sql-vpngw/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,762 @@
...
+        },
+        "aadUsername": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of your AAD user or whoever needs access to resources such as Azure SQL."
+            }
+        },
+        "loginUser": {
+            "type": "string",
+            "defaultValue": "appadmin",
```
Comment (bmoore-msft) [inline 101-asev2-appservice-sql-vpngw/azuredeploy.json:25]: remove defaultValue

File: 101-asev2-appservice-sql-vpngw/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,762 @@
...
+        },
+        "loginPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The administrator password of the VM and SQL Server."
+            }
+        },
+        "privateDbDnsZoneName": {
+            "type": "string",
+            "defaultValue": "privatelink.database.windows.net",
```
Comment (bmoore-msft) [inline 101-asev2-appservice-sql-vpngw/azuredeploy.json:44]: use the environment() function to get the uri

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Pull request merged**